tests/integration/object/user/userkeys/userkeys.go

```diff
@@ -0,0 +1,837 @@
+			assert.Equal(t, string(liveSecret.Data["SecretKey"]), k.SecretKey)
+		})
+
+		t.Run(fmt.Sprintf("cephObjectStoreUser %q .status.keys is unset", osu1.Name), func(t *testing.T) {
+			liveOsu, err := k8sh.RookClientset.CephV1().CephObjectStoreUsers(ns.Name).Get(ctx, osu1.Name, metav1.GetOptions{})
+			require.NoError(t, err)
+
+			require.NotNil(t, liveOsu.Status)
+			assert.Len(t, liveOsu.Status.Keys, 0)
+		})
+
+		// test transition automatic secret creation -> explicit keys
+		t.Run(fmt.Sprintf("add keys to CephObjectStoreUser %q", osu1.Name), func(t *testing.T) {
+			liveOsu, err := k8sh.RookClientset.CephV1().CephObjectStoreUsers(ns.Name).Get(ctx, osu1.Name, metav1.GetOptions{})
+			require.NoError(t, err)
+
+			liveOsu.Spec.Keys = []cephv1.ObjectUserKey{
+				{
+					AccessKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret1.Name,
+						},
+						Key: "AWS_ACCESS_KEY_ID",
+					},
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret1.Name,
+						},
+						Key: "AWS_SECRET_ACCESS_KEY",
+					},
+				},
+				{
+					AccessKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret2.Name,
+						},
+						Key: "AWS_ACCESS_KEY_ID",
+					},
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret2.Name,
+						},
+						Key: "AWS_SECRET_ACCESS_KEY",
+					},
+				},
+				{
+					AccessKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret3.Name,
+						},
+						Key: "AWS_ACCESS_KEY_ID",
+					},
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret3.Name,
+						},
+						Key: "AWS_SECRET_ACCESS_KEY",
+					},
+				},
+				{
+					AccessKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret4.Name,
+						},
+						Key: "AWS_ACCESS_KEY_ID",
+					},
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret4.Name,
+						},
+						Key: "AWS_SECRET_ACCESS_KEY",
+					},
+				},
+				{
+					AccessKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret5.Name,
+						},
+						Key: "AWS_ACCESS_KEY_ID",
+					},
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: secret5.Name,
+						},
+						Key: "AWS_SECRET_ACCESS_KEY",
+					},
+				},
+			}
+
+			_, err = k8sh.RookClientset.CephV1().CephObjectStoreUsers(ns.Name).Update(ctx, liveOsu, metav1.UpdateOptions{})
+			require.NoError(t, err)
+		})
+
+		{
+			secrets := []*corev1.Secret{secret1, secret2, secret3, secret4, secret5}
```

Comment:
For as test complexity/maintenance, is it possible to scale this down to 2 or 3 secrets? I would imagine that 2-3 is sufficient to test things end-to-end without trivializing the feature. 

For example, I think 2 secrets above at setup, and 3 here would be fine.

Reply:
The context is that I started off with 4 test secrets. 2 to exist when a CephObjectStoreUser is created and another 2 to confirm batch replacing the keys work.  Then I decided I wanted prove to myself there wasn't an issue with using more than 2 keys, so I changed the initial CR creation to 3. That brought the total number of test secrets which need to be created to 5. At this point in the test script, all the secrets already exist anyways.  We could use fewer but it with only remove chars from the two lines which declare the slices of test secrets and not remove any lines of code.

Reply:
Also, until yesterday the total number of keys being set was used to detect if reconcile had occurred but the detection logic is now much more rigorous.